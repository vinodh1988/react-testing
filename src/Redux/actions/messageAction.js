
 export const MessageAction=()=>{
      let messages=["React JS is  componenent based","Redux makes react better",
    "Redux is common library","React has states and props","Angular is also component based"];

    return{
        type: "MESSAGE_ACTION",
        data: messages[2]
    }
 }