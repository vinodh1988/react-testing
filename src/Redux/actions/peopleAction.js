import Axios from 'axios';

export const peopleAction=(people)=>{
    
    return{
        type:"PEOPLE_ACTION",
        data:people
    }
}

export const peopleRead=()=>{
     return (dispatch)=>{
         Axios.get("http://ec2-3-16-166-243.us-east-2.compute.amazonaws.com:8896/api/people")
         .then(
     
         (response)=>dispatch(peopleAction(response.data)),
         ()=>dispatch(peopleAction([]))
         );
  }
}

export const peopleWrite=(person)=>{
    console.log(person);
    return (dispatch)=>{
Axios.post("http://ec2-3-16-166-243.us-east-2.compute.amazonaws.com:8896/api/people",person)
.then(
     
         (response)=>dispatch(peopleRead()),
         ()=>dispatch(peopleRead())
         );
    }
}