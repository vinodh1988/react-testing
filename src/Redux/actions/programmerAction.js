
export const addDeveloper=(developer)=>{
    return{
        type:"DEV_ACTION",
        data: developer
    }
}