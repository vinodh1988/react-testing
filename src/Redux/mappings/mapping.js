import {combineReducers,bindActionCreators} from 'redux';
import  {developerReducer,messageReducer,peopleReducer} from '../reducers/messageReducer';
import {connect} from 'react-redux';
import {MessageAction} from '../actions/messageAction';
import {addDeveloper} from '../actions/programmerAction';
import {peopleWrite} from '../actions/peopleAction';

import  MessageBox from '../../MessageBox';
import  Programmers from '../../Programmers';
import People from '../../People';

export const state=combineReducers({
    messages:messageReducer,
    developersData: developerReducer,
    peopleData: peopleReducer
})

const mapStoreToPropsMessage=(state)=>{
    return {message: state.messages.message}
}

const mapActionToPropsMessage=(dispatch)=>{
    return bindActionCreators({updateMessage:MessageAction},dispatch);
}

//Programmer mapping
const mapStoreToPropsProgrammers=(state)=>{
    return {programmers: state.developersData.developers}
}

const mapActionToPropsProgrammer=(dispatch)=>{
    return bindActionCreators({addProgrammer:addDeveloper},dispatch);
}

//People Mapping
const mapStoreToPropsPeople=(state)=>{
    return {people: state.peopleData.people}
}

const mapActionToPropsPeople=(dispatch)=>{
    return bindActionCreators({storePerson:peopleWrite},dispatch);
}


export const MessageRedux=connect(mapStoreToPropsMessage,mapActionToPropsMessage)(MessageBox);
export const ProgrammerRedux=connect(mapStoreToPropsProgrammers,mapActionToPropsProgrammer)(Programmers)
export const PeopleRedux=connect(mapStoreToPropsPeople,mapActionToPropsPeople)(People)


