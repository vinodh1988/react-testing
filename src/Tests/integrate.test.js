import configureStore from 'redux-mock-store';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

const mockStore = configureStore([]);
import {MessageRedux} from '../Redux/mappings/mapping';
import {MessageAction} from '../Redux/actions/messageAction'

describe('My Connected React-Redux Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
        messages:{message:"World is big"}
      });

      store.dispatch = jest.fn();

      component = renderer.create(
        <Provider store={store}>
          <MessageRedux />
        </Provider>);
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should dispatch an action on button click', () => {
    renderer.act(() => {
      component.root.findByType('button').props.onClick();
    });
    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(MessageAction());
    
  });


});