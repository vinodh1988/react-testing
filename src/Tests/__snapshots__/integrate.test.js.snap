// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  style={
    Object {
      "backgroundColor": "lightgreen",
      "fontFamily": "Arial",
      "height": "300px",
      "margin": "10px",
      "padding": "10px",
      "textAlign": "center",
      "width": "600px",
    }
  }
>
  <h3>
    World is big
  </h3>
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Get New Message
  </button>
</div>
`;
